tikv-client/src/main/java/com/pingcap/tikv/expression/visitor/ProtoConverter.java

```diff
@@ -0,0 +1,282 @@
+    builder.setSig(protoSig);
+    return builder.build();
+  }
+
+  protected Expr visit(ArithmeticBinaryExpression node, Void context) {
+    // assume after type coerce, children should be compatible
+    Expression child = node.getLeft();
+    String typeSignature = getTypeSignature(child);
+    ScalarFuncSig protoSig;
+    switch (node.getCompType()) {
+      // TODO: Add test for bitwise push down
+      case BIT_AND:
+        protoSig = ScalarFuncSig.BitAndSig;
+        break;
+      case BIT_OR:
+        protoSig = ScalarFuncSig.BitOrSig;
+        break;
+      case BIT_XOR:
+        protoSig = ScalarFuncSig.BitXorSig;
+        break;
+      case DIVIDE:
+        protoSig = ScalarFuncSig.valueOf("Divide" + typeSignature);
+        break;
+      case MINUS:
+        protoSig = ScalarFuncSig.valueOf("Minus" + typeSignature);
+        break;
+      case MULTIPLY:
+        protoSig = ScalarFuncSig.valueOf("Multiply" + typeSignature);
+        break;
+      case PLUS:
+        protoSig = ScalarFuncSig.valueOf("Plus" + typeSignature);
+        break;
+      default:
+        throw new TiExpressionException(String.format("Unknown comparison type %s", node.getCompType()));
+    }
+    Expr.Builder builder = scalaToPartialProto(node, context);
+    builder.setSig(protoSig);
+    return builder.build();
+  }
+
+  @Override
+  protected Expr visit(ComparisonBinaryExpression node, Void context) {
+    // assume after type coerce, children should be compatible
+    Expression child = node.getLeft();
+    String typeSignature = getTypeSignature(child);
+    ScalarFuncSig protoSig;
+    switch (node.getComparisonType()) {
+      case EQUAL:
+        protoSig = ScalarFuncSig.valueOf("EQ" + typeSignature);
+        break;
+      case GREATER_EQUAL:
+        protoSig = ScalarFuncSig.valueOf("GE" + typeSignature);
+        break;
+      case GREATER_THAN:
+        protoSig = ScalarFuncSig.valueOf("GT" + typeSignature);
+        break;
+      case LESS_EQUAL:
+        protoSig = ScalarFuncSig.valueOf("LE" + typeSignature);
+        break;
+      case LESS_THAN:
+        protoSig = ScalarFuncSig.valueOf("LT" + typeSignature);
+        break;
+      case NOT_EQUAL:
+        protoSig = ScalarFuncSig.valueOf("NE" + typeSignature);
+        break;
+      default:
+        throw new TiExpressionException(String.format("Unknown comparison type %s", node.getComparisonType()));
+    }
+    Expr.Builder builder = scalaToPartialProto(node, context);
+    builder.setSig(protoSig);
+    return builder.build();
+  }
+
+  @Override
+  protected Expr visit(ColumnRef node, Void context) {
+    Expr.Builder builder = Expr.newBuilder();
+    builder.setTp(ExprType.ColumnRef);
+    CodecDataOutput cdo = new CodecDataOutput();
+    // After switching to DAG request mode, expression value
+    // should be the index of table columns we provided in
+    // the first executor of a DAG request.
+    IntegerCodec.writeLong(cdo, node.getColumnInfo().getOffset());
```

Comment:
There's one PR's logic omitted here. More details can be found here https://github.com/pingcap/tikv-client-lib-java/pull/203/files#diff-24cc1d9198b06e4fdc4ab8ca532df54aR82